Source/VaultSharp.Extensions.Configuration/VaultChangeWatcher.cs

```diff
@@ -54,26 +54,34 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
 
             while (!stoppingToken.IsCancellationRequested)
             {
-                await Task.Delay(TimeSpan.FromSeconds(minTime), stoppingToken).ConfigureAwait(false);
-                if (stoppingToken.IsCancellationRequested)
-                {
-                    break;
-                }
-
-                for (var j = 0; j < this._configProviders.Count(); j++)
+                try
                 {
-                    var timer = timers[j];
-                    timer -= minTime;
-                    if (timer <= 0)
+                    await Task.Delay(TimeSpan.FromSeconds(minTime), stoppingToken).ConfigureAwait(false);
+                    if (stoppingToken.IsCancellationRequested)
                     {
-                        this._configProviders.ElementAt(j).Load();
-                        timers[j] = this._configProviders.ElementAt(j).ConfigurationSource.Options.ReloadCheckIntervalSeconds;
+                        break;
                     }
-                    else
+
+                    for (var j = 0; j < this._configProviders.Count(); j++)
                     {
-                        timers[j] = timer;
+                        var timer = timers[j];
+                        timer -= minTime;
+                        if (timer <= 0)
+                        {
+                            this._configProviders.ElementAt(j).Load();
+                            timers[j] = this._configProviders.ElementAt(j).ConfigurationSource.Options
+                                .ReloadCheckIntervalSeconds;
+                        }
+                        else
+                        {
+                            timers[j] = timer;
+                        }
                     }
                 }
+                catch (Exception ex)
```

Comment:
Can we be more specific regarding exceptions here, e.g. I don't want to log TaskCancelledException as it's expected and I want to break the while cycle only?  E.g. I think there are limited set of exception we actually want to log and mute.

Reply:
@MrZoidberg, I agree. The exception can be more specific. I didn't quite get you when you  said 

> I want to break the while cycle only?

Is there a situation where the while loop might need to be broken in the case of an exception? Won't that bring the background service to a halt? Looking forward to your comments.

Reply:
I think in case of TaskCancelledException it's safe to break the while, but not raise exception, just close the worker. In cases when the exception is advertised and expected in code e.g. timeout - just log, in all other cases pass the exception, e.g. OutOfMemoryException

Reply:
From experience, TaskCancelledException is being thrown by the http client if the timeout period of 100 seconds is exceeded. Would breaking the while still be favourable in this case given that the worker should have a chance to try load the configurations after sleeping for a while?

Reply:
ok got it.